src/fs.h

```diff
@@ -140,6 +140,21 @@ static inline path PathFromString(const std::string& string)
     return std::filesystem::path(string);
 #endif
 }
+
+/**
+ * Create directory (and if necessary its parents), unless the leaf directory
+ * already exists or is a symlink to an existing directory.
+ * This is a temporary workaround for an issue in libstdc++ that has been fixed
+ * upstream [PR101510].
+ */
+static inline bool create_directories(const std::filesystem::path& p)
+{
+    if (fs::is_symlink(p) && fs::is_directory(p)) {
```

Comment:
I could be confused but it reads as though the comment says "or" and the logic says "and"
```suggestion
    if (fs::is_symlink(p) || fs::is_directory(p)) {
```

Reply:
Right, `or` would work, but it's not necessary. This `if` statement is to catch the specific situation that is buggy, it doesn't aim to implement anything further.

Reply:
Or just `std::filesystem::exists()` would also work, I guess but will have a wider scope.

> catch the specific situation that is buggy

+1

Reply:
What if it is a symlink to a file, a device, etc, instead of a directory? I think it should still fail in that case. That's why I didn't use the more general `exists`.

Reply:
By "fail" you mean to throw an exception or to return `false`? According to the spec, it should return `false` in that case.

Reply:
OK. Yeah thanks, but I think it's fine to deviate from the spec in this case, it's fatal in our case.